### PR TITLE
Modular Auth services

### PR DESCRIPTION
**Pull request by davidyaha**
This is a work in progress PR to reflect current status and invite contributors to advance it as I am currently not able to do so. (For at least two weeks probably..)

Basic gist is that I am refactoring our main client/server packages to include only "session management" code with the general "authorize" method implemented on the various authorization services.

This is based on the work done by @pradel at the [next](https://github.com/js-accounts/next) repo.

Please comment here before taking this further ahead.

Thanks!

### CODE REVIEW

**Comment by Aetherall** [comment]
is this ready for a try ?

**Comment by davidyaha** [comment]
Nope. Mid work.. Just pushed as I done this a couple of weeks ago and wanted to reflect that as you were asking for current status. 
I've now changed the base branch to our previous `typescript-next` branch to so you can go over just the changes related to implementing the modular auth.

I think that I've stopped with something working but I truly won't have the time to put into it in the next couple of weeks.

I'll help if you have any questions so don't hesitate to comment here.

**Comment by pradel** [comment]
I think this is ready for review

**Comment by pradel** [comment]
See the https://github.com/js-accounts/examples rest-example folder (I pushed on master sorry)

**Comment by TimMikeladze** [comment]
@davidyaha @pradel Is this good to merge? We can address any fixes with additional PRs.

**Comment by davidyaha** [comment]
Writing tests for oauth and we're good. I'll try to finish that today.

File: packages/oauth/__tests__/accounts-oauth.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
+import { UserObjectType } from '@accounts/common';
+import { DBInterface } from '@accounts/server';
+import { AccountsOauth } from '../src';
+
+const user: UserObjectType = {
+  id: '1',
+  username: 'neo',
+  email: 't1@matrix.com',
+};
+const mockStore: DBInterface = {
```
Comment (TimMikeladze) [inline packages/oauth/__tests__/accounts-oauth.ts]: I think it be useful to move `mockStore` to a shared location to be reused by other tests in other packages. Could be left for a future PR.

**Comment by davidyaha** [comment]
@pradel @TimMikeladze Two  issues that I can see on the current state of this PR. 
1. We don't have client tests enabled
2. Our hooks functionality is missing.

Would love some help on these :)

**Comment by dotansimha** [comment]
@davidyaha @pradel 

what is missing for merging this PR?

**Comment by dotansimha** [comment]
@davidyaha can we merge it? something is missing?

File: packages/client/src/accounts-client.ts [hunk 2]
```diff
@@ -314,8 +311,6 @@ export class AccountsClient {
       throw new AccountsError('Username or Email is required');
     }
 
-    const hashAlgorithm = this.options.passwordHashAlgorithm;
```
Comment (davidyaha) [inline packages/client/src/accounts-client.ts]: Do we lose client side encryption by doing so?

**Pull request merged**